MT-1742-Update-SDK

sebastian-krajna opened the pull request:
- Added missing standard events
- Added support for new Appsflyer properties

sebastian-krajna committed:
- Introduced a new event, "locationcoordinates", to the AppsFlyerConstants enum,

sebastian-krajna committed:
Refactor AppsFlyerConstants and AppsFlyerInstance to enhance settings management

- Updated AppsFlyerConstants to include new settings for advertising identifier and IDFV collection, along with deprecation notes for existing settings.
- Modified AppsFlyerInstance to handle the new settings, ensuring backward compatibility with deprecated options.
- Enhanced the example configuration in appsflyer.json to reflect the updated settings structure.

sebastian-krajna committed:
Added support for missing properties

sebastian-krajna committed:
Enhance AppsFlyer integration with new features and settings

- Updated AppsFlyerInstance to accept a URL for the facebookDeferredAppLink, improving data handling.
- Added new properties and commands in appsflyer.json for host configuration, currency setting, phone number tracking, and deep link resolution.
- Implemented UI actions in various view controllers to allow users to set host, currency, phone number, and resolve deep links, enhancing user interaction.
- Expanded unit tests to cover new functionalities, ensuring robust testing of the AppsFlyer integration.

sebastian-krajna committed:
Update storyboards to enhance UI for AppsFlyer integration

- Adjusted layout dimensions in Activities, Ecommerce, and Main storyboards to accommodate new UI elements.
- Added buttons for setting currency and phone number in Ecommerce and Main storyboards, respectively.
- Introduced a new button in Activities for setting the AppsFlyer host and a tracking toggle for user interaction.
- Implemented a new TravelViewController in Activities storyboard for booking travel, including fields for travel details and a segmented control for travel class selection.

sebastian-krajna reviewed (commented):
In this PR I reviewed the missing properties of the Appsflyer class
https://dev.appsflyer.com/hc/docs/ios-sdk-reference-appsflyerlib#properties

and also I added missing standard events 
https://dev.appsflyer.com/hc/docs/in-app-events-ios

Enricoza commented on Sources/AppsFlyerInstance.swift:
```diff
@@ -26,13 +26,17 @@ public protocol AppsFlyerCommand {
-    public override init() { }
+    public override init() {
+        super.init()
+        AppsFlyerLib.shared().delegate = self
```
What is the purpose of adding this line here?

sebastian-krajna replied:
oh I see now, delegate doesn't do anything without tealium, so it's pointless

Enricoza commented on Sources/AppsFlyerInstance.swift:
```diff
@@ -86,6 +101,28 @@ public class AppsFlyerInstance: NSObject, AppsFlyerCommand {
+        if let appInviteOneLinkID = settings[AppsFlyerConstants.Settings.appInviteOneLinkID] as? String {
+            appsFlyer.appInviteOneLinkID = appInviteOneLinkID
+        }
+        if let deepLinkTimeout = settings[AppsFlyerConstants.Settings.deepLinkTimeout] as? Int {
+            appsFlyer.deepLinkTimeout = UInt(deepLinkTimeout)
+        }
```
We should check that deeplink is greater than or equal to 0 or this could crash
```suggestion
        if let deepLinkTimeout = settings[AppsFlyerConstants.Settings.deepLinkTimeout] as? Int,
           deepLinkTimeout >= 0 {
            appsFlyer.deepLinkTimeout = UInt(deepLinkTimeout)
        }
```

sebastian-krajna replied:
hmm, since we talked about the logic should be in VendorRemoteCommand, so I created a similar thing there

Enricoza commented on Sources/AppsFlyerConstants.swift:
```diff
@@ -46,34 +46,48 @@ public enum AppsFlyerConstants {
+        case setCustomerId = "setcustomerid" // customerUserID
         case disableTracking = "disabletracking"
-        case resolveDeepLinkUrls = "resolvedeeplinkurls"
+        case resolveDeepLinkUrls = "resolvedeeplinkurls" // DEPRECATED: use resolve_deep_link_urls in settings instead. Will be removed in a future release.
```
This should not be deprecated, even if we decide to add the new config option

Enricoza commented on Sources/AppsFlyerConstants.swift:
```diff
@@ -46,34 +46,48 @@ public enum AppsFlyerConstants {
         static let debug = "debug"
-        static let disableAdTracking = "disable_ad_tracking"
-        static let disableAppleAdTracking = "disable_apple_ad_tracking"
+        static let disableAdTracking = "disable_ad_tracking" // DEPRECATED: disables both IDFA and IDFV collection. Use disable_advertising_identifier and disable_idfv_collection instead. Will be removed in a future release.
```
Don't deprecate it. And don't provide the more granular options until some customer requests it.

Enricoza commented on Sources/AppsFlyerConstants.swift:
```diff
@@ -46,34 +46,48 @@ public enum AppsFlyerConstants {
+        static let disableIDFVCollection = "disable_idfv_collection"
+        static let disableAppleAdsAttribution = "disable_apple_ads_attribution"
+        static let disableAppleAdTracking = "disable_apple_ad_tracking" // DEPRECATED: disables SKAdNetwork attribution. Use disable_skadnetwork instead. Will be removed in a future release.
+        static let disableSKAdNetwork = "disable_skadnetwork"
```
Don't rename this, there's no need.

Enricoza commented on Tests/MockAppsFlyerInstance.swift:
```diff
@@ -101,6 +105,15 @@ class MockAppsFlyerInstance: AppsFlyerCommand {
+    func disableAppleAdsAttribution(_ disable: Bool) {
+        disableAppleAdsAttributionCount += 1
+        lastDisableAppleAdsAttribution = disable
+    }
```
Where is this used?

Enricoza commented on Tests/AppsFlyerRemoteCommandTests.swift:
```diff
@@ -330,4 +330,26 @@ class AppsFlyerRemoteCommandTests: XCTestCase {
+            "settings": settings
+        ]
+        appsFlyerCommand.processRemoteCommand(with: payload)
+        XCTAssertEqual(appsFlyerInstance.lastSettings?["disable_apple_ads_attribution"] as? Bool, true)
```
Why did you test only this one specifically and not all the other settings that were added?

Enricoza commented on Sources/AppsFlyerInstance.swift:
```diff
@@ -86,6 +101,28 @@ public class AppsFlyerInstance: NSObject, AppsFlyerCommand {
+        }
+        if let resolveDeepLinkURLs = settings[AppsFlyerConstants.Settings.resolveDeepLinkURLs] as? [String] {
+            appsFlyer.resolveDeepLinkURLs = resolveDeepLinkURLs
+        }
```
There's no need to add this settings option. Customers are already setting it with the command. Let's keep it simple for now.

Enricoza reviewed (changes requested):
Looks good so far, just a few changes required.

sebastian-krajna committed:
- Updated AppsFlyerFramework version to 6.17.2 in Cartfile.resolved and Podfile.lock.
- Modified publish.sh to improve script execution by adding a shebang and quoting variables for better safety.
- Enhanced xcframeworks.sh by updating the codesigning process to use a certificate variable instead of a hardcoded team name.
- Cleaned up deprecated settings in AppsFlyerConstants and AppsFlyerInstance, ensuring backward compatibility while removing unused properties.
- Added validation for deepLinkTimeout in AppsFlyerRemoteCommand to filter out negative values.
- Expanded unit tests to cover new settings.

Enricoza committed:
Fix trailing whitespaces and indentation
